### PR TITLE
WIP chapter 2

### PR DESCRIPTION
**Pull request by jowharshamshiri**
Translated the introduction to chapter 2. This is a first draft to just get in sync with other contributors and find our common tone in translation. Also created glossary.csv in the root of 'fa' folder to standardize translations of some commonly mistranslated words--this too is a first draft open to suggestions from other contributors.

Related to #49

### CODE REVIEW

File: chapters/fa/chapter2/1.mdx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#مقدمه
```
Comment (lewtun) [inline chapters/fa/chapter2/1.mdx:1]: To render this on the website, we should add a `_toctree.yml` file with the following content:

```
- title: 1. Transformer models # Translate this
  sections:
  - local: chapter2/1
    title: Introduction # And this
```

File: chapters/fa/chapter2/2.mdx [hunk 2]
```diff
@@ -0,0 +1,353 @@
+<FrameworkSwitchCourse {fw} />
```
Comment (lewtun) [inline chapters/fa/chapter2/2.mdx:1]: You should remove all the untranslated files - we only render on the website the chapters / sections that are translated

**Review by MehranMirkhan — APPROVED** [review]
I am a native Persian speaker. The translations look good to me.

**Comment by jowharshamshiri** [comment]
@lewtun can we get this through I have another page incoming and need to add terms to the glossary but don't want to invalidate this review. :)

**Comment by jowharshamshiri** [comment]
I have chapter 0 ready for merge in a separate branch on my fork. Will submit in separate PR for review.

File: chapters/fa/glossary/1.mdx [hunk 3]
```diff
@@ -0,0 +1,52 @@
+| معادل در منبع         | معادل در ترجمه   |
```
Comment (lewtun) [inline chapters/fa/glossary/1.mdx:1]: Similarly, here we'll need a H1 header with something like

```
# Glossary <= translate this
```

**Comment by lewtun** [comment]
> @lewtun can we get this through I have another page incoming and need to add terms to the glossary but don't want to invalidate this review. :)

Hey @jowharshamshiri happy to merge this once we can get the docs to build. I've left some suggestions on trying this yourself, but please let me know if you need some help!

File: chapters/fa/chapter2/1.mdx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<div dir="rtl">
+<h1>مقدمه</h1>
```
[the remark this answers is not in the document]
Comment (jowharshamshiri) [reply chapters/fa/chapter2/1.mdx:2]: Thanks for this. will do tonight. I am having so much trouble both with the characters and the RTL nature of the language. Finding editors is hard too. We might need to write that guide, haven't come across one myself.

**Comment by jowharshamshiri** [comment]
Hey @lewtun, I experimented with DISABLE-FRONTMATTER-SECTIONS and also adding an empty code block but alas to no avail. I want to open an issue on doc-builder, whereby we add a 'direction' arg or otherwise automatically figure out the direction of languages. That info is then used to automatically handle the direction of the generated divs and whatnot. If we do that we can dispense with the extra RTL divs and spans everywhere in Persian and also other RTL-language docs. Is that something that sounds worthwhile to you or am I confused?

File: chapters/fa/glossary.csv [hunk 5]
```diff
@@ -0,0 +1,32 @@
+Transformer,ترنسفورمر
```
[the remark this answers is not in the document]
Comment (lewtun) [reply chapters/fa/glossary.csv]: LGTM, thanks!

**Comment by lewtun** [comment]
Thanks a lot for diving into this @jowharshamshiri - I had no idea it would be so tricky 😬 . 

I think opening an issue on `doc-builder` would make a lot of sense as I can already see we're running into the same issue with other RTL languages. I suggest providing a super simple example of an MDX file and corresponding `_toctree.yml` file that reproduces the problem and separately shows the hacks you've needed to implement to make it work. 

In the meantime, I'll test I can build the current PR!

File: .github/workflows/build_documentation.yml [hunk 6]
```diff
@@ -14,6 +14,6 @@ jobs:
       package: course
       path_to_docs: course/chapters/
       additional_args: --not_python_module
-      languages: en es fr he pt ru th tr
+      languages: en es fa fr he pt ru th tr
```
Comment (lewtun) [inline .github/workflows/build_documentation.yml:17]: FYI I've added Farsi to the list of languages here so we can build the docs in the CI

**Comment by jowharshamshiri** [comment]
Hey @lewtun, while trying to file the issue and reproduce the problem I found an simpler workaround which I included in new commits. Please kindly merge. The first header now renders correctly as RTL. I filed an issue on doc-builder [here](https://github.com/huggingface/doc-builder/issues/176) to automate this.

P.S. I think maybe in the CI build you have the lang arg still set to en. It is building the original English docs.

**Comment by lewtun** [comment]
Thank you for finding a nice fix! Will merge this now 🤗 

> P.S. I think maybe in the CI build you have the lang arg still set to en. It is building the original English docs.

Yeah that's a current limitation with our UI - we need to add a dropdown button to view the PR in language X, hopefully will have that soon!

**Pull request merged**